lib/pengine/rules.c

```diff
@@ -124,15 +131,22 @@ pe_test_expression(xmlNode *expr, GHashTable *node_hash, enum rsc_role_e role,
                    crm_time_t *now, crm_time_t *next_change,
                    pe_match_data_t *match_data)
 {
-    pe_rule_eval_data_t rule_data = {
-        .node_hash = node_hash,
+    pcmk_rule_input_t rule_input = {
         .now = now,
-        .match_data = match_data,
-        .rsc_data = NULL,
-        .op_data = NULL
+        .node_attrs = node_hash,
     };
 
-    return pe_eval_subexpr(expr, &rule_data, next_change);
+    if (match_data != NULL) {
```

Comment:
Either that, or create a temporary old rule data object, set its match data to `match_data`, and map it.